src/Settings/Math.ts

```diff
@@ -28,6 +28,25 @@ export function GenerateMathSettings(): IMathDecidrCustoms {
 
                 return Math.round(average / actors.length);
             },
+            "averageLevelWildPokemon": function (constants: IMathConstants, equations: IMathEquations, options: IWildPokemonSchema[]): number {
+                let average: number = 0;
+
+                for (const wildPokemon of options) {
+                    if (wildPokemon.level) {
+                        average += wildPokemon.level * wildPokemon.rate;
+                    } else {
+                        let levelAverage: number = 0;
+
+                        for (const level of wildPokemon.levels) {
+                            levelAverage += level * (1 / wildPokemon.levels.length);
+                        }
+
+                        average += levelAverage * wildPokemon.rate;
```

Comment:
I think you could combine this whole `else` block using `map` and maybe even `reduce` logic to make it a little cleaner.